idb/postgres/postgres.go

```diff
@@ -2169,7 +2189,11 @@ func (db *IndexerDb) AssetBalances(ctx context.Context, abq idb.AssetBalanceQuer
 	if len(whereParts) > 0 {
 		query += " WHERE " + strings.Join(whereParts, " AND ")
 	}
-	query += " ORDER BY addr ASC"
+	if abq.Address != nil {
+		query += " ORDER BY assetid ASC"
```

Comment:
perhaps this should be unconditional?

Reply:
there are existing code paths to look up all the balances for a specific asset that don't set this.